__tests__/__snapshots__/ord.test.js.snap

```diff
@@ -65,11 +65,11 @@ exports[`Tests for default ORD document Successfully create ORD Documents with d
       "extensible": {
         "supported": "no",
       },
-      "ordId": "capire.bookshopordsample:apiResource:undefined.CatalogService:v1",
+      "ordId": "capire.bookshopordsample:apiResource:CatalogService:v1",
       "partOfGroups": [
-        "sap.cds:service:capire.bookshopordsample:undefined.CatalogService",
+        "sap.cds:service:capire.bookshopordsample:CatalogService",
       ],
-      "partOfPackage": "capirebookshopordsample:package:undefined:v1",
+      "partOfPackage": "capire.bookshopordsample:package:customer.capirebookshopordsample:v1",
```

Comment:
Sorry, I'm still confused. Is this the namespace as it was provided in the cdsrc file? I think it would be better to go with a `customer.bookshopsample` as it's more realistic.

Reply:
This is the namespace as it is given in the package.json because in the bookshop test folder, we don't have .cdsrc.json file.
As you know, we are constructing namespace this way:
`cds.env["ord"]?.namespace || packageName.replace(/[^\w/]/g, "").replace(/\//, ".");`

Again, there is no .cdsrc.json file in the ORD/__tests__/bookshop folder, so the first part of OR statement is discarded;

Reply:
ok, but if we use the package name, we need to prepent it with `customer.` (as you started doing when we talked about it)

Reply:
```
cds.env["ord"]?.namespace || 'customer.` + packageName.replace(/[^\w/]/g, "").replace(/\//, "");
```

Reply:
This is how the packages looks like now in our test: 
```
  packages: [
    {
      ordId: "customer.capire.bookshopordsample:package:capirebookshopordsample:v1",
      title: "capire bookshop ord sample",
      shortDescription: "Short description for capire bookshop ord sample",
      description: "Description for capire bookshop ord sample",
      version: "1.0.0",
      partOfProducts: [
        "customer:product:capire.bookshop.ord.sample:",
      ],
      vendor: "customer:vendor:Customer:",
    },
  ],
```
@zongqichen  @Fannon : Looks correct now?

Reply:
no, there should not be `.capire` in the namespace, we also need to escape `.` away unfortunately. Correct would be: `customer.capirebookshopordsample`

Reply:
When we take the package name and make a default namespace out of it, we basically need to replace any non ASCII char with "". The result will be really ugly, that's why I hope that they choose their own namespaces explicitly

Reply:
Hi @Fannon, why should we use the value from package.json? If there are muli packages generated, each package OrdId will be same? Could you explain more about this [sentences](https://pages.github.tools.sap/CentralEngineering/open-resource-discovery-specification/spec-v1#ord-id):
```
<resourceName> := the technical resource name.
MUST only contain ASCII letters (a-z, A-Z), digits (0-9) and the special characters -, _ and ..
```
1. It shows special characters `-` `_` `.` are allowed, but we use regex to filter out?
2. What's the meaning of technical resource name?

Reply:
> It shows special characters - _ . are allowed, but we use regex to filter out?

there are different rules for the namespaces than for the resource name. They each need different escaping rules. For example `.` is allowed in a Resource name, but cannot be allowed in a namepace because a `.` separates namespaces fragments from each other.

Reply:
Looks good! :)